#! /usr/bin/env python
# Task Long configuration

PACKAGE='sim_tasks'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *
from task_manager_lib.parameter_generator import *

gen = TaskParameterGenerator()
#       Name                    Type            Description
#       Default    Min   Max
gen.add("angle",         double_t, 0,    "Reference angle between the boat and the shore",  1.57)
gen.add("ang_velocity",  double_t, 0,    "Angular velocity",  0.5)
gen.add("angle_error",   double_t, 0,    "Angle error accepted with the reference angle",  0.1)
gen.add("angle_range",   double_t, 0,    "Range on each side of the reference angle, where we detect the error_angle",  0.78)
gen.add("dist_threshold",double_t, 0,    "Distance maximum of detection of the shore",  30.0)
gen.add("max_lin_vel",   double_t, 0,    "Maximum linear velocity",  0.3)
gen.add("max_ang_vel",   double_t, 0,    "Maximum angular velocity",  0.5)

exit(gen.generate(PACKAGE, "sim_tasks", "TaskAlignWithShore"))

